apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/regionscoped/HypervisorDetails.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.nova.v2_0.domain.regionscoped;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.json.SerializedNames;
+
+/**
+ * Hypervisor which shows id, name and other details
+ */
+@AutoValue
+public abstract class HypervisorDetails {
+
+   public abstract String getId();
+
+   public abstract String getName();
+
+   public abstract int getCurrentWorkload();
```

Comment:
good point.
Took the reference from here. added nullable annotation and made fields objects for nullable properties.
https://github.com/openstack/nova/blob/master/nova/objects/compute_node.py